src/main/java/org/folio/entlinks/controller/delegate/AuthorityArchiveServiceDelegate.java

```diff
@@ -54,21 +58,45 @@ public AuthorityFullDtoCollection retrieveAuthorityArchives(Integer offset, Inte
   @Transactional(readOnly = true)
   public void expire() {
     var retention = fetchAuthoritiesRetentionDuration();
-
     if (retention.isEmpty()) {
       return;
     }
-
     var tillDate = LocalDateTime.now().minusDays(retention.get());
-    try (Stream<AuthorityArchive> archives = authorityArchiveRepository.streamByUpdatedTillDate(tillDate)) {
+    if (isConsortiumTenant()) {
+      consortiumTenantProcess(tillDate);
+    } else {
+      tenantProcess(tillDate);
+    }
+  }
+
+  private boolean isConsortiumTenant() {
+    var consortiumTenants = tenantsService.getConsortiumTenants(context.getTenantId());
```

Comment:
The timer triggers the "/authority-storage/expire/authorities" endpoint for the central tenant and all member tenants at the same time asynchronously. If a flow is run for a member tenant, the member tenant retrieves all records by retention period and deletes them without checking whether it is a shadow copy or a local record.
In this case, the flow for the central tenant simultaneously tries to delete the same shadow copy from the same member tenant, and the transaction fails.
That's why at the beginning of the flow we need to determine if this tenant is a member tenant and prevent the member tenant from deleting shadow copies. The member tenant should only delete local authority records.

Reply:
Yes, you are right. We can avoid checking if this member is a member tenant. One SQL query can be used for all tenants, since only a member tenant can have the "CONSORTIUM-" prefix in the records.